shared/actions/teams.js

```diff
@@ -394,38 +394,51 @@ const _getDetails = function*(action: TeamsGen.GetDetailsPayload): Saga.SagaGene
     }, {})
 
     const infos = []
-    const types = ['admins', 'owners', 'readers', 'writers']
-    const typeMap = {
-      admins: 'admin',
-      owners: 'owner',
-      readers: 'reader',
-      writers: 'writer',
+    const types: Types.TeamRoleType[] = ['reader', 'writer', 'admin', 'owner']
+    const typeToKey: Types.TypeMap = {
+      reader: 'readers',
+      writer: 'writers',
+      admin: 'admins',
+      owner: 'owners',
     }
     types.forEach(type => {
-      const members = details.members[type] || []
+      const key = typeToKey[type]
+      const members = details.members[key] || []
       members.forEach(({active, fullName, username}) => {
         infos.push([
           username,
           Constants.makeMemberInfo({
             active,
             fullName,
-            type: typeMap[type],
+            type,
             username,
           }),
         ])
       })
     })
 
-    const invites = map(details.annotatedActiveInvites, invite =>
-      Constants.makeInviteInfo({
+    const invites = map(details.annotatedActiveInvites, (invite: RPCTypes.AnnotatedTeamInvite) => {
+      const role = Constants.teamRoleByEnum[invite.role]
```

Comment:
using `map` basically makes flow give up on typing `invite`, so need to explicitly type it (which necessitated the changes below)